contracts/dao/pre-propose/cwd-pre-propose-single-overrule/src/contract.rs

```diff
@@ -103,34 +80,236 @@ pub fn execute(
                 funds: vec![],
             });
 
+            let subdao_name = get_subdao_name(&deps, &subdao_address)?;
+            let prop_name: String = format!(
+                "Reject the proposal #{} of the '{}' subdao",
+                proposal_id, subdao_name
+            );
+            let prop_desc: String = format!(
+                "If this proposal will be accepted, the DAO is going to \
+overrule the proposal #{} of '{}' subdao (address {})",
+                proposal_id, subdao_name, subdao_address
+            );
+
             let internal_msg = ExecuteInternal::Propose {
                 msg: ProposeMessageInternal::Propose {
                     // Fill in proposer based on message sender.
                     proposer: Some(info.sender.to_string()),
-                    title: "Overrule proposal".to_string(),
-                    description: "Reject the decision made by subdao".to_string(),
+                    title: prop_name,
+                    description: prop_desc,
                     msgs: vec![overrule_msg],
                 },
             };
 
-            let result = PrePropose::default().execute(deps, env, info, internal_msg);
+            let next_proposal_id = &get_next_proposal_id(&deps)?;
 
-            match result {
-                Ok(response) => Ok(response),
-                Err(error) => Err(PreProposeOverruleError::PreProposeBase(error)),
-            }
+            PROPOSALS.save(
+                deps.storage,
+                (proposal_id, timelock_contract_addr),
+                next_proposal_id,
+            )?;
+
+            Ok(internal_msg)
         }
+        // The following messages are forwarded to the base contract
+        ExecuteMsg::ProposalCreatedHook {
+            proposal_id,
+            proposer,
+        } => Ok(ExecuteInternal::ProposalCreatedHook {
+            proposal_id,
+            proposer,
+        }),
+        ExecuteMsg::ProposalCompletedHook {
+            proposal_id,
+            new_status,
+        } => Ok(ExecuteInternal::ProposalCompletedHook {
+            proposal_id,
+            new_status,
+        }),
+        // ExecuteMsg::Withdraw and ExecuteMsg::UpdateConfig are unsupported
+        // ExecuteMsg::Withdraw is unsupported because overrule proposals should have no deposits
+        // ExecuteMsg::UpdateConfig since the config has only the info about deposits,
+        // no custom fields are added.
         _ => Err(PreProposeOverruleError::MessageUnsupported {}),
+    };
+    PrePropose::default()
+        .execute(deps, env, info, internal_msg?)
+        .map_err(PreProposeOverruleError::PreProposeBase)
+}
+
+fn get_subdao_from_timelock(
+    deps: &DepsMut,
+    timelock_contract: &String,
+) -> Result<Addr, PreProposeOverruleError> {
+    let timelock_config: TimelockTypes::Config = deps.querier.query_wasm_smart(
+        timelock_contract.to_string(),
```

Comment:
already &String

Reply:
Changed the argument type to Addr for consistency reasons